packages/cli-core/src/package-manager-controller/package_manager_controller.ts

```diff
@@ -112,6 +120,45 @@ Get started by running \`${this.binaryRunner} amplify sandbox\`.`;
     );
   }
 
+  /**
+   * runWithPackageManager - Factory function that runs a command with the specified package manager's binary runner
+   */
+  runWithPackageManager(
+    args: string[] = [],
+    dir: string,
+    options?: {
+      env?: Record<string, string>;
+      stdin?: 'inherit' | 'pipe' | 'ignore';
+      stdout?: 'inherit' | 'pipe' | 'ignore';
+      stderr?: 'inherit' | 'pipe' | 'ignore';
+      extendEnv?: boolean;
+    }
+  ): ExecaChildProcess<string> {
+    return this.executeWithDebugLogger(
+      dir,
+      this.binaryRunner,
+      args,
+      this.execa,
+      options
+    );
+  }
+
+  getPackageManagerCommandArgs = (
```

Comment:
this should go back to the cdk deployer. The only thing here that the package manager controller needs to expose is something like `getCdkAppCommand` which returns `'${this.binaryRunner} tsx ${backendLocator.locate()}'`. Then that can be plugged into the rest of the CDK args by the cdk deployer.